obspy/clients/nrl/client.py

```diff
@@ -0,0 +1,258 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+'''
+obspy.clients.nrl.client.py for navigating, and picking responses from the
+Nominal Response Library.
+
+:copyright:
+    Lloyd Carothers IRIS/PASSCAL, 2016
+:license:
+    GNU Lesser General Public License, Version 3
+    (http://www.gnu.org/copyleft/lesser.html)
+'''
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA
+
+import codecs
+import io
+import os
+import requests
+import sys
+
+if sys.version_info.major == 2:
+    import ConfigParser as configparser
+else:
+    import configparser
+
+from obspy.core.inventory.response import Response
+from obspy.io.xseed import Parser
+
+
+class NRL(object):
+    """
+    NRL client base class for accessing the Nominal Response Library:
+    http://ds.iris.edu/NRL/
+
+    Created with a URL for remote access or filesystem accessing a local copy.
+    """
+    _index = 'index.txt'
+    def __new__(cls, root=None):
+        try:
+            o = requests.utils.urlparse(root)
+        except AttributeError as e:
+            # root is None
+            o = None
+
+        if root is None or o.scheme == 'http':
+            # Create RemoteNRL
+            return super(NRL, cls).__new__(RemoteNRL)
+        elif os.path.isdir(o.path):
+            # Create LocalNRL
+            return super(NRL, cls).__new__(LocalNRL)
```

Comment:
If you invert the if/elif part, i.e. if isdir(o.path), else: then it would also work with `https`. You could also just pass everything that is not a directory to the RemoteNRL class and see if it can handle it.